pkg/cliplugins/kubestellar/get_log/cmd/cmd.go

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/client-go/util/homedir"
+	"k8s.io/klog/v2"
+)
+
+var (
+	kubeconfig string
+	followFlag bool
+	linesFlag  string
+	whichArg   string
+	logger     klog.Logger
+)
+
+func New(streams genericclioptions.IOStreams) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "get-log [flags] which",
+		Short: "Get the log output from a central KubeStellar or kcp process",
+		Args:  cobra.ExactArgs(1),
+		Run:   getLog,
+	}
+
+	fs := pflag.NewFlagSet("get-log", pflag.ExitOnError)
+	klog.InitFlags(flag.CommandLine)
+	fs.AddGoFlagSet(flag.CommandLine)
+	cliOpts := genericclioptions.NewConfigFlags(true)
+	cliOpts.AddFlags(fs)
+	fs.StringVarP(&kubeconfig, "kubeconfig", "c", "", "Path to the kubeconfig file")
+	fs.BoolVarP(&followFlag, "follow", "f", false, "Follow the log output")
+	fs.StringVarP(&linesFlag, "lines", "n", "+0", "Number of lines to display")
+	fs.Parse(os.Args[1:])
```

Comment:
The command line parsing should not be done as part of Command construction.
I suspect that a client of cobra is expected to hook into the flag framework in cobra, and cobra will invoke the Parse function at the right time.

Reply:
The parsing should not be done as part of the Command construction.
The baseline design of cobra is that cobra invokes the parsing before invoking the Command's Run or RunE method.